anti_patterns/deny-warnings.md

```diff
@@ -0,0 +1,35 @@
+# `#![deny(warnings)]`
+
+## Description
+
+A well-intentioned crate author wants to ensure their code builds without
+warnings. So they annotate their crate root with
+
+## Example
+
+```rust
+#![deny(warnings)]
+
+// All is well.
+```
+
+## Drawbacks
+
+Until, say, the language changes, and as part of the transition, a lint is
+introduced that `warn`s, and Boom! The code no longer builds. Or someone tries
+to compile the code with additional lints (e.g. [clippy]), only to find out
+that they have to remove the annotation first.
+
+## Alternatives
+
+Our plucky developer hero should have used
+`RUSTFLAGS="-D warnings" cargo build"` to build their crate, removing
+the need for the annotation in their code.
+
```

Comment:
> It is probably worth stating the alternative of denying specific warnings as attributes which does not have the behaviour you describe.

Is this true in general? Even when only specific warnings are chosen, the meanings of the warnings themselves can change over time. That was part of the motivation for rust-lang/rfcs#1193, after all.